lib/services/livesync/livesync-service.ts

```diff
@@ -34,16 +37,61 @@ export class LiveSyncService implements ILiveSyncService {
 
 			let projectDir = this.$project.getProjectDir().wait();
 
-			let livesyncData = {
+			let livesyncData: ILiveSyncData = {
 				platform: platform,
 				appIdentifier: this.$project.projectData.AppIdentifier,
 				projectFilesPath: projectDir,
 				syncWorkingDirectory: projectDir,
-				excludedProjectDirsAndFiles: this.excludedProjectDirsAndFiles
+				excludedProjectDirsAndFiles: this.excludedProjectDirsAndFiles,
+				additionalConfigurations: this.$project.projectInformation.configurations
 			};
 
-			$liveSyncServiceBase.sync(livesyncData).wait();
+			let deviceConfigurationInfos: Mobile.IApplicationInfo[] = [];
 
+			let configurations = this.$project.getConfigurationsSpecifiedByUser();
+			if (!configurations.length) {
+				this.$devicesService.execute(device => (() => {
```

Comment:
I'll go ahead and extract it as a method, however calling it inside the `if` and respectively the `else` branch is an optimization. In the `if` the call is made using the original livesync information, whereas in the else multiple calls are made with configuration-specific livesync information